airdrop/application/services/airdrop_services.py

```diff
@@ -130,14 +123,77 @@ def get_airdrop_window_stake_details(self, inputs):
 
         return status, response
 
-    def get_stake_info(self, staking_contract_address, address):
-        contract = get_contract_instance(
-            STAKING_CONTRACT_PATH, staking_contract_address, contract_name='STAKING')
+    def get_stake_info(self, contract_address, user_wallet_address, airdrop_rewards):
+        try:
+
+            is_stake_window_is_open, max_stake_amount = self.get_stake_window_details(
+                contract_address)
+
+            is_user_can_stake, already_staked_amount = self.get_stake_details_of_address(
+                contract_address, user_wallet_address)
+
+            is_stakable, stakable_amount, tranfer_to_wallet = self.get_stake_and_claimable_amounts(
+                airdrop_rewards, is_stake_window_is_open, max_stake_amount, is_user_can_stake, already_staked_amount)
 
-        is_stakable, amount, rewards_computation_index, bonus_amount = contract.functions.getStakeInfo(
-            address).call()
+            return is_stakable, stakable_amount, tranfer_to_wallet
+        except BaseException as e:
+            raise e("Exception on get_stake_info {}".format(e))
+
+    def get_stake_and_claimable_amounts(self, airdrop_rewards, is_stake_window_is_open, max_stake_amount, is_user_can_stake, already_staked_amount):
+        # User can stake if stake window is open and user can stake
+        is_stakable = True if is_user_can_stake and is_stake_window_is_open else False
+        tranfer_to_wallet = 0
+        stakable_amount = 0
+
+        if is_stakable:
+            allowed_amount_for_stake = max_stake_amount - already_staked_amount
+            if(airdrop_rewards <= allowed_amount_for_stake):
+                # If airdrop rewards is less than allowed amount for stake then stake the full airdrop rewards
+                stakable_amount = airdrop_rewards
+            else:
+                stakable_amount = allowed_amount_for_stake
+
+            # Amount user can claim to wallet after staking
+            tranfer_to_wallet = airdrop_rewards - stakable_amount
+        else:
+            tranfer_to_wallet = airdrop_rewards
+
+        return is_stakable, stakable_amount, tranfer_to_wallet
+
+    def get_stake_details_of_address(self, contract_address, user_wallet):
+        try:
+            contract = get_contract_instance(
+                STAKING_CONTRACT_PATH, contract_address, contract_name='STAKING')
 
-        return is_stakable, amount
+            stake_info = contract.functions.getStakeInfo(user_wallet).call()
```

Comment:
If user has not staked at all will stake_info still have two elements?

Reply:
Yes, It return False & 0 if not staked @raamb